lib/bamboo/mailer.ex

```diff
@@ -184,8 +184,7 @@ defmodule Bamboo.Mailer do
   end
 
   @doc false
-  def parse_opts(mailer, opts) do
-    otp_app = Keyword.fetch!(opts, :otp_app)
+  def parse_opts(mailer, otp_app) do
     config = Application.fetch_env!(otp_app, mailer) |> Enum.into(%{})
```

Comment:
Since you're working on the config, could you change `Enum.into(%{})` into `Map.new`? That is faster and I think a bit more clear.